inventory/internal/store/util.go

```diff
@@ -292,3 +293,91 @@ func GetOrderByOptions[T OrderOption](orderBy string, columnValidator func(strin
 	}
 	return opts, nil
 }
+
+// ValidateOSMetadata verifies that the given JSON encoded metadata contains the required fields, otherwise returns an error.
+func ValidateOSMetadata(metadata string) (string, error) {
+	if metadata == "" {
+		return "", nil
+	}
+
+	if err := validateFlatJSONNoDuplicates([]byte(metadata)); err != nil {
+		return "", err
+	}
+	var tmp map[string]string
+	err := json.Unmarshal([]byte(metadata), &tmp)
+	if err != nil {
+		zlog.InfraSec().InfraErr(err).Msgf("Error while un-marshaling the OS metadata")
+		return "", errors.Wrap(err)
+	}
+
+	// From OSMetadata to list of Metadata for easier validation
+	metaList := make([]Metadata, 0, len(tmp))
+	for k, v := range tmp {
+		metaList = append(metaList, Metadata{
+			Key:   k,
+			Value: v,
+		})
+	}
+
+	// validate the actual key and value fields in metadata
+	metaerr := validateKeyValue(metaList)
+	if metaerr != nil {
+		zlog.InfraSec().InfraErr(metaerr).Msgf("Error while validating the OS metadata")
+		return "", errors.Wrap(metaerr)
+	}
+	// Marshal the metadata, so what we store is always the same, not random spaces in between.
+	val, err := json.Marshal(tmp)
+	if err != nil {
+		zlog.InfraSec().InfraErr(err).Msgf("Error while re-marshaling the OS metadata")
+		return "", errors.Wrap(err)
+	}
+	return string(val), nil
+}
+
+func validateFlatJSONNoDuplicates(data []byte) error {
```

Comment:
do we really need to write all this code? I mean is there not a json validator already that we can re-use?